lightning/src/ln/channel.rs

```diff
@@ -7470,9 +7474,17 @@ impl<SP: Deref> Writeable for Channel<SP> where SP::Target: SignerProvider {
 					htlc_id.write(writer)?;
 					err_packet.write(writer)?;
 				}
-				&HTLCUpdateAwaitingACK::FailBlindedHTLC { ref htlc_id } => {
-					3u8.write(writer)?;
+				&HTLCUpdateAwaitingACK::FailMalformedHTLC {
+					htlc_id, failure_code, sha256_of_onion
+				} => {
+					// We don't want to break downgrading by adding a new variant, so write a dummy
+					// `::FailHTLC` variant and write the real malformed error as an optional TLV.
+					malformed_htlcs.push((htlc_id, failure_code, sha256_of_onion));
+
+					let dummy_err_packet = msgs::OnionErrorPacket { data: Vec::new() };
+					2u8.write(writer)?;
 					htlc_id.write(writer)?;
+					dummy_err_packet.write(writer)?;
 				}
```

Comment:
No, downgraded nodes will use this dummy htlc data to fail this htlc backwards. This prevents them from dropping the malformed HTLCs (which are written as optional TLVs) on `read` post-downgrade, which would cause them to not fail backwards and eventually be FC'd on.